mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td

```diff
@@ -812,7 +812,8 @@ def Tosa_MulOp : Tosa_Op<"mul", [
   let arguments = (ins
     Tosa_Tensor:$input1,
     Tosa_Tensor:$input2,
-    Optional<TosaTensorRankOf<[Tosa_Int8], [1]>>:$shift
+    // Apply right shift on i32_t input data only
+    Tosa_ScalarInt8Tensor:$shift
```

Comment:
Why shall this be required at dialect level? Not in favor of a compulsory parameters when it doesn't make sense. Any good reasoning behind this?